Semantic Scholar.js

```diff
@@ -37,148 +37,423 @@
+}
+
+function detectWeb(doc, url) {
+	if (doc.querySelector('#app')) {
 		Z.monitorDOMChanges(
 			doc.querySelector('#app'),
 			{ childList: true, subtree: true }
 		);
-		return false;
 	}
-	let citation = ZU.xpathText(doc, '//pre[@class="bibtex-citation"]');
-	let type = citation.split('{')[0].replace('@', '');
-	return bibtex2zoteroTypeMap[type];
-}
 
-function getSearchResults(doc, checkOnly) {
-	var items = {};
-	var found = false;
-	var rows = doc.querySelectorAll('.result-page .cl-paper-row > a');
-	for (let row of rows) {
-		let href = row.href;
-		let title = ZU.trimInternal(row.textContent);
-		if (!href || !title) continue;
-		if (checkOnly) return true;
-		found = true;
-		items[href] = title;
+	if (url.includes('semanticscholar.org/search?')) {
+		const hasSearchResults = getSearchResults(doc, true);
+		if (hasSearchResults) {
+			return 'multiple';
+		}
 	}
-	return found ? items : false;
+
+	return getItemTypeFromBibtex(doc);
 }
 
 function doWeb(doc, url) {
-	if (detectWeb(doc) == 'multiple') {
+	const docType = detectWeb(doc, url)
+	if (docType == 'multiple') {
 		Zotero.selectItems(getSearchResults(doc, false), (items) => {
-			if (items) ZU.processDocuments(Object.keys(items), parseDocument);
+			if (items) {
+				ZU.processDocuments(Object.keys(items), scrape);
+			}
 		});
+		return;
 	}
-	else if (url.includes('pdfs.semanticscholar.org')) {
-		let urlComponents = url.split('/');
-		let paperId = urlComponents[3] + urlComponents[4].replace('.pdf', '');
-		const API_URL = 'https://api.semanticscholar.org/';
-		ZU.processDocuments(API_URL + paperId, parseDocument);
-	}
-	else {
-		parseDocument(doc, url);
-	}
-}
 
-function parseDocument(doc, url) {
-	let citation = ZU.xpathText(doc, '//pre[@class="bibtex-citation"]');
-	let type = citation.split('{')[0].replace('@', '');
-	const itemType = bibtex2zoteroTypeMap[type];
+	scrape(doc, url);
+}
 
-	var item = new Zotero.Item(itemType);
+function scrape(doc, url) {
+	const itemType = getItemTypeFromBibtex(doc);
+	const item = new Zotero.Item(itemType);
 
-	// load structured schema data
-	const schemaTag = doc.querySelector("script.schema-data");
+	const schemaTag = doc.querySelector('script.schema-data');
 	const schemaObject = JSON.parse(schemaTag.innerHTML);
-	const article = schemaObject["@graph"][1][0];
-
+	const article = schemaObject['@graph'][1][0];
+	
 	item.title = article.name;
-	item.abstractNote = article.description;
+	item.abstractNote = article.abstract;
+	item.date = article.datePublished;
+	item.url = url;
+	item.attachments.push({
+		url: url,
+		title: url.includes('semanticscholar.org/reader') ? 'Semantic Reader Link' : 'Semantic Scholar Link',
+		mimeType: 'text/html',
+		snapshot: false
+	});
+
+	if (article.about) {
+		item.abstractNote = article.abstract.substring(article.about.length + 1);
+		item.notes.push('[TLDR] ' + article.about);
+	}
+
+	if (itemType == 'conferencePaper' && article.publisher) {
+		item.publisher = article.publisher.name;
```

Comment:
Thanks for the suggestion. I've tried the DOI Content Negotiation translator and it looks like it fetches more paper details. But I am also seeing that the `libraryCatalog` field would be substituted to `DOI.org` instead of `Semantic Scholar`. Would it be appropriate if call out this translator first when there is a DOI found in a paper, then assign `Semantic Scholar` to the `libraryCatalog` attribute and create "Semantic Scholar"-related attachments?